open_instruct/dpo_tune.py

```diff
@@ -544,22 +546,13 @@ def load_model():
     starting_epoch = 0
 
     # Potentially load in the weights and states from a previous save
-    if args.resume_from_checkpoint:
-        if args.resume_from_checkpoint is not None or args.resume_from_checkpoint != "":
-            checkpoint_path = args.resume_from_checkpoint
-            path = os.path.basename(args.resume_from_checkpoint)
-        else:
-            # Get the most recent checkpoint
-            dirs = [f.name for f in os.scandir(os.getcwd()) if f.is_dir()]
-            dirs.sort(key=os.path.getctime)
-            path = dirs[-1]  # Sorts folders by date modified, most recent checkpoint is the last
-            checkpoint_path = path
-            path = os.path.basename(checkpoint_path)
-
-        accelerator.print(f"Resumed from checkpoint: {checkpoint_path}")
-        accelerator.load_state(path)
+    last_checkpoint_path = get_last_checkpoint_path(args)
+    if last_checkpoint_path:
+        accelerator.print(f"Resumed from checkpoint: {last_checkpoint_path}")
+        accelerator.load_state(last_checkpoint_path)
```

Comment:
Interesting! I did not know `accelerator.load_state` loads all the states (model, optimizer, scheduler)

Reply:
Sometimes there may be too much magic going on. At some point we could prob verify with a picture like 

<img width="572" alt="image" src="https://github.com/user-attachments/assets/85dc4c01-419b-41a3-94a8-bf07489e3298">

Reply:
Yeah, Wandb testing is on my agenda. From digging into the DPO stuff, it appears because the reference model loading is hacked in a bit, so the accelerate state doesnt pick it up (this is nice for us, tho, since the reference model is fine).

Reply:
Resume from checkpoint for DPO may work best with Log Prob cacheing @hamishivi @vwxyzjn